pkg/services/provisioning/provisioning.go

```diff
@@ -78,7 +79,9 @@ func (ps *provisioningServiceImpl) Run(ctx context.Context) error {
 
 		// Wait for unlock. This is tied to new dashboardProvisioner to be instantiated before we start polling.
 		ps.mutex.Lock()
-		pollingContext, cancelFun := context.WithCancel(ctx)
+		// Using background here because otherwise if root context was canceled the select later on would
+		// non-deterministically take one of the route possibly going into one polling loop before exiting.
+		pollingContext, cancelFun := context.WithCancel(context.Background())
```

Comment:
@bergquist Not sure if this is a good way to solve this. I did not find a way to check if `ctx.Done()` is closed in case `case <-pollingContext.Done():` will be taken first after root context is canceled.

Reply:
:+1: